### PR TITLE
Fix API docs CSS files so dark mode works

### LINKED ISSUE #111941: dev/docs/assets/overrides.css is breaking dark mode
Go to https://master-api.flutter.dev

Turn on dark mode by clicking this button:

Screen Shot 2022-09-19 at 8 09 48 PM

Observe the page doesn't change.

If you delete the `link` element to the overrides.css, it works.

We probably just need to update that file to be compatible with darkmode.

@gspencergoog might know more about the history of why we have that and how to safely update it.

/cc @kevmoo

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This fixes the https://master-api.flutter.dev site so that dark mode works properly.

It also re-enables the link button on samples so that you can deep link to a sample (the button was always there, but somehow got disabled and turned the same color as the background).

I've also adjusted the colors and fonts to be a bit closer to those of the main flutter.dev website.

Before:
Screen Shot 2022-09-20 at 4 59 23 PM

Fixed Dark Mode:
Screen Shot 2022-09-20 at 4 59 07 PM

Fixed Light Mode:
Screen Shot 2022-09-20 at 4 59 14 PM

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/111941

## Tests
 - Fixed sanity test for snippets to be a little more lenient.

### CODE REVIEW

File: dev/docs/assets/overrides.css [hunk 1]
```diff
@@ -1,37 +1,11 @@
...
-  font-size: 42px !important;
-  letter-spacing: -1px;
-}
-
-h2 {
-  color: #111;
-  font-size: 24px;
+/* Prefer Google Sans to bring the API docs closer to the main site's look */
+body {
+  font-family: "Google Sans", "Roboto", sans-serif;
```
Comment (Hixie) [inline dev/docs/assets/overrides.css:5]: `Google Sans` is intended only for headers, not body text.

**Comment by Hixie** [comment]
test-exempt: configuration change
...though it would be really nice to have some sort of actual end-to-end test for the docs, just so we catch it when we accidentally break search or whatever.

**Review by Hixie — APPROVED** [review]
RSLGTM

I admit I didn't look super closely at all the CSS changes.

File: dev/docs/assets/overrides.css (hunk 1 above)
Comment (gspencergoog) [reply dev/docs/assets/overrides.css:14]: Ahh, good point.  OK, I revamped it to only apply to headers.

**Pull request merged**